### ISSUE TITLE
Cookie handling on HTTP 401 broken

### ISSUE THREAD

**Issue by MikeN123** [OfficeDev/ews-java-api#188]
With HttpClient 3, we used to handle cookie headers on a 401 response. This changed in HttpClient 4, see this ticket in the HttpClient JIRA: https://issues.apache.org/jira/browse/HTTPCLIENT-1451

One of the issues this causes is that if EWS is published through ISA/TMG server, authentication fails after 5 minutes (the default credential cache time). Strictly speaking, this is a TMG bug, but since TMG is EOL we cannot expect anything there. TMG sets a forms-based auth (FBA) cookie, which is invalidated after 5 minutes. TMG then sends a 401 response, with a header to remove the FBA cookie, but HttpClient does not process this header. It automatically attempts reauthentication, which fails, because the cookie still exists.

tl;dr: if using EWS behind TMG, it is not possible to reuse the ExchangeService for more than 5 minutes, due to the Cookie headers not being processed.

We're looking for a solution, PR will follow if we have one.

**Comment by serious6** [comment]
_ref. #192 due to disconnect of ExchangeService_

**Comment by MikeN123** [comment]
I'm not sure what you are implying? #192 is not really a question that should be asked here, since you would first need to put EWS behind CAS before you can think about the API accessing EWS (and no clue how you would do that, EWS expects Basic/SPNEGO auth afaik, or OAuth in O365).

**Comment by serious6** [comment]
ok. maybe I just saw a connection where no one exists.
